src/main/java/net/lldp/checksims/token/TokenList.java

```diff
@@ -160,4 +166,14 @@ public String toString() {
     public int hashCode() {
         return type.hashCode() ^ super.hashCode();
     }
+
+    public int numValid() {
```

Comment:
I imagine it's to ensure there will never be overflow issues if it's called on a very large stream? Streams are very high performance, so there's no reason you couldn't have one over 2^31 elements in size.